<launch>
    <arg name="use_sim"           default="true"/>
    <arg name="urdf_param_name"   default="/wx250s/robot_description"/>
    <arg name="chain_root"        default="wx250s/base_link"/>
    <arg name="chain_tip"         default="wx250s/ee_gripper_link"/>
    <arg name="input_pose_topic"  default="/kiosk/arm/js/axes"/>
    <arg name="node_start_delay"  default="1.0" />

    <include file="$(find interbotix_xsarm_control)/launch/xsarm_control.launch">
      <arg name="robot_model"   value="wx250s"/>
      <arg name="use_sim"       value="$(arg use_sim)"/>
    </include>

    <node name="kdl_interbotix_bridge"  pkg="kdl_arm"  type="kdl_interbotix.py" output="screen" respawn="true"/>

    <group ns='joint_limits'>
      <rosparam file="$(find kdl_arm)/config/wx250s_jl.yaml"/>
    </group>
    <node name="kdl_wx250s" pkg="kdl_arm"  type="main" output="screen" respawn="true" launch-prefix="bash -c 'sleep $(arg node_start_delay); $0 $@' ">
      <param name="urdf_param_name"     type="str"  value="$(arg urdf_param_name)" />
      <param name="chain_root"          type="str"  value="$(arg chain_root)" />
      <param name="chain_tip"           type="str"  value="$(arg chain_tip)" />
      <param name="input_pose_topic"    type="str"  value="$(arg input_pose_topic)" />
    </node>

</launch>
